File: CompilationTest/InheritanceTest/results/InheritanceTest.cs
using System;

public interface IPrinterBase
{
    int SomeBaseFunc();
}

public interface IPrinter: IPrinterBase
{
    void PrintIt();
}

public class BasePrinter: IPrinter
{
    public int NumValue = 42;

    public virtual string GetValue()
    {
        return "Base";
    }
    
    public void PrintIt()
    {
        Console.WriteLine($"BasePrinter: {this.GetValue()}");
    }
    
    public int SomeBaseFunc()
    {
        return this.NumValue;
    }
}

public class ChildPrinter: BasePrinter
{
    public override string GetValue()
    {
        return "Child";
    }
}

public class TestClass
{
    public IPrinter GetPrinter(string name)
    {
        var result = name == "child" ? new ChildPrinter() : new BasePrinter();
        return result;
    }
    
    public void TestMethod()
    {
        var basePrinter = this.GetPrinter("base");
        var childPrinter = this.GetPrinter("child");
        basePrinter.PrintIt();
        childPrinter.PrintIt();
        Console.WriteLine($"{basePrinter.SomeBaseFunc()} == {childPrinter.SomeBaseFunc()}");
        
        var baseP2 = new BasePrinter();
        var childP2 = new ChildPrinter();
        Console.WriteLine($"{baseP2.NumValue} == {childP2.NumValue}");
    }
}

public class Program
{
    static public void Main(string[] args)
    {
        try 
        {
            new TestClass().TestMethod();
        }
        catch (System.Exception e)
        {
            System.Console.WriteLine($"Exception: {e.Message}");
        }
    }
}

File: CompilationTest/InheritanceTest/results/InheritanceTest.swift
protocol IPrinterBase {
    func someBaseFunc() -> Int;
}

protocol IPrinter: IPrinterBase {
    func printIt() -> Void;
}

class BasePrinter: IPrinter {
  var numValue: Int = 42

  func getValue() -> String {
      return "Base"
  }

  func printIt() -> Void {
      print("BasePrinter: \(self.getValue())")
  }

  func someBaseFunc() -> Int {
      return self.numValue
  }
}

class ChildPrinter: BasePrinter {
  override func getValue() -> String {
      return "Child"
  }
}

class TestClass {
  func getPrinter(name: String) -> IPrinter? {
      let result: IPrinter? = name == "child" ? ChildPrinter() : BasePrinter()
      return result
  }

  func testMethod() -> Void {
      let basePrinter: IPrinter? = self.getPrinter(name: "base")
      let childPrinter: IPrinter? = self.getPrinter(name: "child")
      basePrinter!.printIt()
      childPrinter!.printIt()
      print("\(basePrinter!.someBaseFunc()) == \(childPrinter!.someBaseFunc())")
      
      let baseP2: BasePrinter? = BasePrinter()
      let childP2: ChildPrinter? = ChildPrinter()
      print("\(baseP2!.numValue) == \(childP2!.numValue)")
  }
}

TestClass().testMethod()